sdk/nodejs/cmd/run/run.ts

```diff
@@ -172,7 +174,7 @@ export function run(
 
     if (typeScript) {
         tsnode.register({
-            typeCheck: true,
```

Comment:
Rather than another envvar couldn't this be read from tsConfig above? Currently we pull "compilerOptions" out from that but stands to reason we could pull "typeCheck" from there as well.

Reply:
I don't believe `tsConfig` files have a `typeCheck` option, as they are defining how typechecking should be performed. In our case, we definitely want our `tsconfig.json` file to declare how to typecheck, as we still typecheck our monorepo via `tsc`, we just don't want Pulumi to also typecheck.

This is similar to how `ts-node` is often run in production settings with `--transpile-only` enabled

Reply:
Good prompt. I went and double checked what options are valid here and in tsconfig, and your right that typeCheck isn't an option in tsconfig. It also looks like the typeCheck option here is also an old deprecated option and should be replaced with transpileOnly.

Reply:
Oh cool, good find. I accepted the suggested changes